helix-rest/src/test/java/org/apache/helix/rest/metadatastore/accessor/TestZkRoutingDataReader.java

```diff
@@ -42,12 +42,16 @@
 
   @BeforeClass
   public void beforeClass() {
+    ZK_SERVER_MAP.get(ZK_ADDR).getZkClient()
+        .deleteRecursively(MetadataStoreRoutingConstants.ROUTING_DATA_PATH);
     _zkRoutingDataReader = new ZkRoutingDataReader(DUMMY_NAMESPACE, ZK_ADDR, null);
   }
 
   @AfterClass
   public void afterClass() {
     _zkRoutingDataReader.close();
+    ZK_SERVER_MAP.get(ZK_ADDR).getZkClient()
+        .deleteRecursively(MetadataStoreRoutingConstants.ROUTING_DATA_PATH);
```

Comment:
Nit: a good addition to this would be a TestHelper.verify() that checks that the deletion has completed.